### ISSUE TITLE
false positive: warn-unused-ignores checks in different python version blocks

### ISSUE THREAD

**Issue by KotlinIsland** [python/mypy#10813]
This type ignore is valid, if python >= 3.9 the block isn't checked, but unused ignore is still checked and a false positive error is raised.
```py
import sys

if sys.version_info < (3, 9):
    asdfgsdfgb()  # type: ignore
```
https://mypy-play.net/?mypy=latest&python=3.10&flags=warn-unused-ignores&gist=f40e0f6968779e61d62bec4a7b85b27c

**Comment by KotlinIsland** [comment]
also:
```py
import typing
if not typing.TYPE_CHECKING:
    asdfgsdfgb()  # type: ignore
```

**Comment by hauntsaninja** [comment]
Duplicate of #8823
